app/controllers/mixins/ems_common_angular.rb

```diff
@@ -465,19 +488,25 @@ def set_ems_record_vars(ems, mode = nil)
       build_connection(ems, endpoints, mode)
     end
 
-    def get_hostname_from_routes(ems, endpoint_hash, token)
-      return nil unless ems.class.respond_to?(:openshift_connect)
-      endpoint = Endpoint.new(endpoint_hash)
+    # TODO: move to backend
+    def get_hostname_from_routes(ems)
```

Comment:
@moolitayer Can you move this method and all the other `container` specific code in this PR to `ems_container_controller`?